pkg/spoke/addon/configuration.go

```diff
@@ -0,0 +1,159 @@
+package addon
+
+import (
+	"crypto/sha256"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	addonv1alpha1 "github.com/open-cluster-management/api/addon/v1alpha1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	installationNamespaceAnnotation = "installNamespace"
+	registrationAnnotation          = "registrations"
+)
+
+// addOnConfig contains meta information of an addon
+type addOnConfig struct {
+	AddOnName             string
+	InstallationNamespace string
+}
+
+// registrationConfig contains necessary information for addon registration
+type registrationConfig struct {
+	AddOnName             string      `json:"addOnName,omitempty"`
```

Comment:
why have public names on internal types?

Reply:
It has to be public because I want to add json tag on it.